src/App.jsx

```diff
@@ -1,247 +1,118 @@
-              href="#/"
-              className="button is-link is-outlined is-fullwidth"
-            >
-              Reset all filters
-            </a>
-          </div>
-        </nav>
-      </div>
+import usersFromServer from './api/users';
+import categoriesFromServer from './api/categories';
+import productsFromServer from './api/products';
+import { Filters } from './components/Filters';
+import { ProductList } from './components/ProductList';
+
+const products = productsFromServer.map((product) => {
+  const category = categoriesFromServer
+    .find(({ id }) => product.categoryId === id);
+  const user = usersFromServer
+    .find(({ id }) => category.ownerId === id);
+
+  return {
+    ...product,
+    category,
+    user,
+  };
+});
+
+function filterProducts(productsToFilter, queries) {
+  const {
+    userId,
+    searchQuery,
+    selectedCategories,
+  } = queries;
+
+  let filteredData = filterByUser(productsToFilter, userId);
+
+  filteredData = filterBySearchQuery(filteredData, searchQuery);
+  filteredData = filterByCategories(filteredData, selectedCategories);
+
+  return filteredData;
+}
+
+function filterByUser(productsToFilter, userId) {
+  if (!userId) {
+    return productsToFilter;
+  }
+
+  return productsToFilter.filter(({ user }) => user.id === userId);
+}
+
+function filterBySearchQuery(productsToFilter, searchQuery) {
+  const lowerQuery = searchQuery.toLowerCase();
+
+  return productsToFilter
+    .filter(({ name }) => name.toLowerCase().includes(lowerQuery));
+}
+
+function filterByCategories(productsToFilter, categories) {
+  if (!categories.length) {
+    return productsToFilter;
+  }
+
+  return productsToFilter
+    .filter(({ category }) => categories.includes(category.id));
+}
+
+export const App = () => {
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedCategories, setSelectedCategories] = useState([]);
+
+  const filterParams = {
+    userId: selectedUser,
+    searchQuery,
+    selectedCategories,
+  };
+
+  const filteredProducts = filterProducts(products, filterParams);
+
+  const addSelectedCategory = (newCategory) => {
+    setSelectedCategories(prevState => [...prevState, newCategory]);
+  };
+
+  const removeSelectedCategory = (categoryToRemove) => {
+    setSelectedCategories(prevState => prevState
+      .filter(id => id !== categoryToRemove));
+  };
+
+  const changeCategory = (id) => {
+    if (selectedCategories.includes(id)) {
+      removeSelectedCategory(id);
+
+      return;
+    }
+
+    addSelectedCategory(id);
+  };
+
+  const clearCategories = () => setSelectedCategories([]);
+
+  return (
+    <div className="section">
+      <div className="container">
+        <h1 className="title">Product Categories</h1>
+
+        <Filters
+          users={usersFromServer}
+          selectedUser={selectedUser}
+          changeSelectedUser={newUser => setSelectedUser(newUser)}
```

Comment:
you can simplify it
```suggestion
          changeSelectedUser={setSelectedUser}
```

Reply:
I wanted to do state lifting here, so that it was obvious where our state changes